### ISSUE TITLE
Porting constrainedby modifiers from the original definition through the redeclare chain does not work

### ISSUE THREAD

**Issue by adrpo** [OpenModelica/OpenModelica#7457]
### Description

While flattening model `TransiEnt.Producer.Gas.Electrolyzer.Check.Test_400A_Espinosa_L2`
you get the error: 
```
[Test_400A_Espinosa_L2Total_original.mo:1379:4-1379:71:writable] 
Error: Function parameter vleFluidPointer was not given by the function call, and does not have a default value.

```
The total model is attached.

### Steps to Reproduce

```
loadModel("Test_400A_Espinosa_L2Total_original.mo"); getErrorString();
instantiateModel(TransiEnt.Producer.Gas.Electrolyzer.Check.Test_400A_Espinosa_L2); getErrorString();

```
### Expected Behavior

Flattens it fine

[Test_400A_Espinosa_L2Total_original.zip](https://github.com/OpenModelica/OpenModelica/files/6458286/Test_400A_Espinosa_L2Total_original.zip)

**Comment by mimeissner** [comment]
I'm not 100% sure, if this is the correct issue, but in the ThermoFluidStream Library we have also experienced, that modifiers from the constraining model are not arriving at a redeclared a model. Our workaround for now is to remove final from the modifiers in the constraining model and include all modifiers when redeclaring. 


I attached a minimal example, with models A,B,C and D. The bahavior I would expect is, that when simulating D, D->a2.k is 200, the same as C->a2.k is, when running C. 

After testing I also found that modifiers from the original model (not the constraining model) are not handed down when the model is redeclared. (see C->a3.k and D->a3.k, which should both be 300, but D->a3.k is 1). 

[testRedeclareModifiers.txt](https://github.com/OpenModelica/OpenModelica/files/7604271/testRedeclareModifiers.txt)

**Comment by casella** [comment]
Let's see what @perost thinks about it.

**Comment by perost** [comment]
It might be the same issue as in #8092.

**Comment by perost** [comment]
With the fixes in #8249 the model now flattens if I also disable the checks for multiple algorithm sections in functions.

**Comment by casella** [comment]
> With the fixes in #8249 the model now flattens if I also disable the checks for multiple algorithm sections in functions.

So I guess this ticket could be closed and one opened about removing multiple algorithm sections in functions in TransiEnt?

**Comment by perost** [comment]
> > With the fixes in #8249 the model now flattens if I also disable the checks for multiple algorithm sections in functions.
> 
> So I guess this ticket could be closed and one opened about removing multiple algorithm sections in functions in TransiEnt?

Yes, the issue with multiple algorithm sections in TransiEnt is that they have partial functions with external sections that are extended by functions that themselves have external sections. It's trivial to have the compiler just ignore one of them, but it's not standard Modelica and should be fixed in the library.

We probably shouldn't open a new issue for that though, since it's not an issue with OpenModelica and TransiEnt does not have issues enabled in their repo. @adrpo is well aware of the issue and is trying to get fixes into TransiEnt, so I'll leave it up to him.
